src/views/SignMessage.vue

```diff
@@ -52,32 +52,33 @@ export default class SignMessage extends Vue {
         if (this.keyguardResult) return;
 
         if (this.request.walletId && this.request.signer) {
-            this.accountSelected(this.request.walletId, this.request.signer.toUserFriendlyAddress());
+            this.accountSelected(this.request.walletId, this.request.signer.toUserFriendlyAddress(), true);
             return;
         }
 
         this.showAccountSelector = true;
-
     }
 
-    private async accountSelected(walletId: string, address: string) {
+    private async accountSelected(walletId: string, address: string, isFromRequest = false) {
         const walletInfo = this.findWallet(walletId);
         if (!walletInfo) {
-            // XXX Should we really return an error here and when checking the address below,
-            // or would it enable malicous sites to query for stored walletIds and addresses?
-            // We could also quietly ignore any unavailable pre-set walletId and address and give
+            // We can also return an error here and when checking the address below,
+            // but it would enable malicous sites to query for stored walletIds and addresses.
+            // Instead we quietly ignore any unavailable pre-set walletId and address and give
             // the user the option to chose as if it was not pre-set.
-            this.$rpc.reject(new Error('WalletId not found'));
+            this.showAccountSelector = true;
+            if (!isFromRequest) throw new Error(`UNEXPECTED: Selected walletId was not found: ${walletId}`);
             return;
```

Comment:
Hm.. I think this error (and the one below) are *so* unexpected, it doesn't make sense to even handle those cases.